### PR TITLE
feat: get offers from BE

### PR DESCRIPTION
**Pull request by mcksyrek**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+import { ServerUrlInterceptor } from './shared/interceptors/server-url.interceptor';
 @NgModule({
   declarations: [AppComponent],
-  imports: [BrowserModule, AppRoutingModule, BrowserAnimationsModule],
-  providers: [],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    BrowserAnimationsModule,
+    HttpClientModule,
```
Comment (ArtemLunt) [inline src/app/app.module.ts:16]: Were probably using the `HttpClient` all over, so pls create some shared module and move it there.

File: src/app/shared/interceptors/server-url.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    request: HttpRequest<unknown>,
+    next: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    request = request.clone({
+      url: environment.serverUrl.concat(request.url),
+    });
+    console.log(request);
+
+    return next.handle(request);
+  }
```
Comment (ArtemLunt) [inline src/app/shared/interceptors/server-url.interceptor.ts:25]: Interceptor is redundant here.
Pls handle it with the [proxy](https://angular.io/guide/build#proxying-to-a-backend-server).

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

File: src/app/shared/interceptors/server-url.interceptor.ts (hunk 2 above)
Comment (mcksyrek) [reply src/app/shared/interceptors/server-url.interceptor.ts:25]: that's schweppes

File: src/app/shared/enums/endpoints.enum.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export enum Endpoints {
+  Offers = '/api/offers',
```
Comment (mcksyrek) [inline src/app/shared/enums/endpoints.enum.ts:2]: I need `/api` to recognize this request by proxy - is there any elegant way to concat it with all API req? or this one is cool?

File: src/app/shared/enums/endpoints.enum.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
 export enum Endpoints {
-  Offers = '/offers',
+  Offers = '/api/offers',
```
Comment (ArtemLunt) [inline src/app/shared/enums/endpoints.enum.ts:2]: Pls rid of this prefix here

File: src/proxy.conf.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "/api": "http://localhost:8080"
```
Comment (ArtemLunt) [inline src/proxy.conf.json:2]: Here you can use `pathRewrite` to add the `/api` prefix and simply map all the requests

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

File: src/app/offers/offers.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  describe('#getOffersList', () => {
+    it('should return an Observable<IOffer[]>', () => {
+      const mockResponse: IOffer[] = [
+        { name: 'mockName1', id: 'mockID1' },
+        { name: 'mockName2', id: 'mockID2' },
+      ];
+
+      service.getOffersList().subscribe(offers => {
```
Comment (ArtemLunt) [inline src/app/offers/offers.service.spec.ts:36]: usually it works in next way:
`spyOn(service, 'method').and.returnValue(mockResponse)`

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

File: src/app/offers/offer-form/offer-form.component.spec.ts [hunk 7]
```diff
@@ -28,32 +30,32 @@ describe('OfferFormComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  describe('submitForm', () => {
+  describe('#submitForm', () => {
     it('should emit formValue', () => {
-      const mockOffer = { offerName: 'testName' };
-      spyOn(component.submitForm, 'emit');
-      component.offerForm.controls.offerName.setValue(mockOffer.offerName);
+      const spy = jest.spyOn(component.submitForm, 'emit');
```
Comment (ArtemLunt) [inline src/app/offers/offer-form/offer-form.component.spec.ts:35]: Pls, provide more meaningful variable name

File: src/app/offers/offers-list/offers-list.component.ts [hunk 8]
```diff
@@ -1,26 +1,30 @@
...
+export class OffersListComponent implements OnInit {
   @Select(OffersState.getOffers)
   readonly offers$: Observable<IOffer[]>;
 
   showOfferForm = false;
 
   constructor(private _store: Store) {}
 
+  ngOnInit(): void {
+    this._store.dispatch(GetOfferListAction);
```
Comment (ArtemLunt) [inline src/app/offers/offers-list/offers-list.component.ts:25]: `GetOfferListAction` is the class. It should be `new GetOfferListAction()`.

File: src/environments/environment.ts [hunk 9]
```diff
@@ -4,6 +4,7 @@
 
 export const environment = {
   production: false,
+  serverUrl: '/api',
```
Comment (ArtemLunt) [inline src/environments/environment.ts:7]: It's not a URL, it's a prefix

**Review by ArtemLunt — APPROVED** [review]

**Pull request merged**